corehq/tests/test_classcleanup.py

```diff
@@ -0,0 +1,142 @@
+from functools import wraps
+from unittest import TestCase
+
+from nose.plugins import PluginTester
+from testil import eq
+
+from .noseplugins import classcleanup as mod
+
+
+class TestClassCleanupPlugin(PluginTester, TestCase):
+    activate = ''  # Activate option not needed. Plugin is always enabled.
+    plugins = [mod.ClassCleanupPlugin()]
+
+    def setUp(self):
+        pass  # super().setUp() is called by self.run_with_errors(...)
+
+    def makeSuite(self):
+        class Test(TestCase):
+            @classmethod
+            @maybe_error(self)
+            def setUpClass(cls):
+                cls.addClassCleanup(lambda: self.result.append("cleanup"))
```

Comment:
Nit: given that 'cleanup' already exists on a test-by-test basis, it might be more intuitive if this was called 'classCleanup'

Reply:
https://github.com/dimagi/commcare-hq/pull/31084/commits/f8e57b130b00e1d47bfdda5843fb3ce4a16021ba